### PR TITLE
Phase2

### LINKED ISSUE #3: You need to add axios to your package.json
You need to add axios to your package.json

_Originally posted by @pchertude34 in https://github.com/aemann2/react-super-store/pull/2#discussion_r596472335_

### LINKED ISSUE #4: The deals should be filtered from the API
The deals should be filtered from the API

_Originally posted by @pchertude34 in https://github.com/aemann2/react-super-store/pull/2#discussion_r596472934_

### LINKED ISSUE #5: The data should be loaded in the page components. Loading the data here will cause the data to be loaded when the user loads the app, regardless if they need it or not. We don't want to load necessary data if we can avoid it. Loading the data here will also make search and pagination much more challenging to implement in the future.
The data should be loaded in the page components. Loading the data here will cause the data to be loaded when the user loads the app, regardless if they need it or not. We don't want to load necessary data if we can avoid it. Loading the data here will also make search and pagination much more challenging to implement in the future.

_Originally posted by @pchertude34 in https://github.com/aemann2/react-super-store/pull/2#discussion_r596474289_

### LINKED ISSUE #6: It probably makes sense to put this, and the other svgs, into a `/assets` or `/icons` file in the src directory.
It probably makes sense to put this, and the other svgs, into a `/assets` or `/icons` file in the src directory.

_Originally posted by @pchertude34 in https://github.com/aemann2/react-super-store/pull/2#discussion_r596475436_

### PR DESCRIPTION
**Pull request by aemann2**
# Phase 2 - Display Items

1. **As a User, I should see a list of items on the "home" page. This should look similar to what is in the prototype.**

- [x] Load data from the "/item/list" route of the Super Store API.
- [x] Create a component to display the item data. This should include PropTypes for prop type checking. It should look something like the cards on the home page of the prototype
- [x] Create a container component to display a list of item components. This component should be used to contain and render the Item cards
- [x] Create a component that renders the rating stars and count.
  - [Full Star SVG](https://guidedprojects.dev/assets/files/star_full-f757f2a6eb46312ac3b0a9f4923add12.svg)
  - [Half Star SVG](https://guidedprojects.dev/assets/files/star-half-c86abc67a14feafe9a821fd5e06e5833.svg)
  - [Empty Star SVG](https://guidedprojects.dev/assets/files/star-empty-849812a23c4515e156571518674ae723.svg)
- [x] _Extra Challenge_: Make cards in the same row always be equal height
- [x] _Extra Challenge_: Align "Add to Cart" Button to the bottom of the card, so all "add to cart" buttons in the row will align.
- [x] _Extra Challenge_: Make the card list responsive for smaller screens. Reduce the number of cards displayed per row from three to two or one depending on the small screen size.

2. **As a User, I should see a list of only on sale items on the "deals" page**

- [x] This page should look and act similar to the home page, but only show items that are on sale. We don't need the search bar on this page
- [x] If there are no on sale items, the user should see a message stating there are no on sale items at this time.

3. **As a User, clicking on the item's title or "add to cart" button will bring me to the Item's product page**

- [x] Create a new React Page and route at "/item/:itemId" where :itemId holds the unique id of the item. This only needs to display the item's id at this point as we will add the data in the next phase

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,36 @@
-import './css/App.css';
+import { useState, useEffect } from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import './css/style.css';
+import axios from 'axios';
```
Comment (pchertude34) [inline src/App.js:3]: You need to add axios to your package.json

File: src/components/pages/Deals.js [hunk 2]
```diff
@@ -1,11 +1,49 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import Item from '../items/Item';
+
+const Deals = (props) => {
+  const { items } = props;
+  // filtering for items on sale
+  const deals = items.filter((item) => item.isOnSale);
```
Comment (pchertude34) [inline src/components/pages/Deals.js]: The deals should be filtered from the API

File: src/components/items/Stars.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import full from './stars/full.svg';
+import half from './stars/half.svg';
+import empty from './stars/empty.svg';
+
+const Stars = (props) => {
+  const { rating } = props;
+
+  const stars = [];
+  // a loop to push stars to the above array based on rating
+  for (let i = stars.length; i < 5; i++) {
```
Comment (pchertude34) [inline src/components/items/Stars.js:12]: Really good job with this!!

File: src/App.js [hunk 4]
```diff
@@ -1,12 +1,36 @@
...
 import Cart from './components/pages/Cart';
-//importing react router
-import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import ItemPage from './components/pages/ItemPage';
 
 function App() {
+  // lifting up the items state so the results can be passed to child pages
+  const [items, setItems] = useState(null);
+
+  useEffect(() => {
```
Comment (pchertude34) [inline src/App.js]: The data should be loaded in the page components. Loading the data here will cause the data to be loaded when the user loads the app, regardless if they need it or not. We don't want to load necessary data if we can avoid it. Loading the data here will also make search and pagination much more challenging to implement in the future.

File: src/components/items/stars/empty.svg [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pchertude34) [inline src/components/items/stars/empty.svg]: It probably makes sense to put this, and the other svgs, into a `/assets` or `/icons` file in the src directory.

File: src/components/items/Stars.js (hunk 3 above)
Comment (aemann2) [reply src/components/items/Stars.js:12]: Thanks, this makes me happy. I was afraid I wasn't using enough finesse with a for-loop, so I'm glad this is an acceptable solution.

File: src/App.js (hunk 1 above)
Comment (aemann2) [reply src/App.js:3]: added in 4ac62df74c5bdab5927a623d702adcd0e14dafe6

File: src/components/items/stars/empty.svg (hunk 5 above)
Comment (aemann2) [reply src/components/items/stars/empty.svg]: added in 4ac62df74c5bdab5927a623d702adcd0e14dafe6

File: src/App.js (hunk 4 above)
Comment (aemann2) [reply src/App.js]: fixed in a9af71b2e0c082692b9402817ed3a70cd11571db

File: src/components/pages/Deals.js (hunk 2 above)
Comment (aemann2) [reply src/components/pages/Deals.js]: fixed in a9af71b2e0c082692b9402817ed3a70cd11571db

**Pull request merged**